runtime/commonMain/src/kotlinx/serialization/Annotations.kt

```diff
@@ -10,60 +10,156 @@ import kotlinx.serialization.json.*
 import kotlin.reflect.*
 
 /**
- * Sets specific serializer for class, property or type argument.
- * When argument is omitted, plugin will generate default implementation inside the class.
+ * The main entry point to the serialization process.
+ * Applying [Serializable] to the Kotlin class instructs the serialization plugin to automatically generate [KSerializer]
```

Comment:
```suggestion
 * Applying [Serializable] to the Kotlin class instructs the serialization plugin to automatically generate implementation of [KSerializer]
```

imo, 'generate [interface]' sounds clumsy